crates/matrix-sdk-base/src/sliding_sync.rs

```diff
@@ -97,7 +97,7 @@ impl BaseClient {
 
         trace!("ready to submit changes to store");
         self.store.save_changes(&changes).await?;
-        self.apply_changes(&changes);
+        self.apply_changes(&changes, true);
```

Comment:
It's not correct. If a `SlidingSync` is configured to run both encryption and has configured lists, then a room that has a change for the latest event + a list change because of a room list operation, will receive two updates.

We should only send the update if the sliding sync is not configured to handle room list operations here.

Can you add a test for this, and fix it please? (I think sending a new message after line 733 in the integration test should cause the double update I'm expecting)

Reply:
If a room has a late decryption that changes the latest_event and at the same time the room is updated by the other code, this will always lead to duplicated updates. It doesn't matter if both updates are made by a single service or two separate services.

I don't think we can fix this without a more involved deduplication algorithm, but this sounds like a rare situation.

Reply:
I think I see: my proposal doesn't work when there's a latest-event change to the room but no changes in the room list. Well, let's nevermind for now, and get back to it later if it turns out to be an issue.